packages/phoenix-event-display/src/tests/managers/three-manager/scene-manager.test.ts

```diff
@@ -0,0 +1,235 @@
+/**
+ * @jest-environment jsdom
+ */
+import { SceneManager } from '../../../managers/three-manager/scene-manager';
+import { Object3D, Scene, Vector3 } from 'three';
+import {
+  AmbientLight,
+  AxesHelper,
+  BoxGeometry,
+  BufferGeometry,
+  Camera,
+  Color,
+  DirectionalLight,
+  Group,
+  Mesh,
+  MeshBasicMaterial,
+} from 'three';
+import { CoordinateHelper } from '../../../helpers/coordinate-helper';
+
+describe('SceneManager', () => {
+  let sceneManager: SceneManager;
+  const ignoreList = [
+    new AmbientLight().type,
+    new DirectionalLight().type,
+    new AxesHelper().type,
+  ];
+
+  beforeEach(() => {
+    sceneManager = new SceneManager(ignoreList, false);
+  });
+
+  it('should update position of directional light for each frame rendered', () => {
+    sceneManager.updateLights(new Camera());
+    expect(sceneManager.cameraLight).toBeUndefined();
+  });
+
+  it('should get a clean copy of the scene', () => {
+    const scene = new Group();
+    scene.add(new Mesh(new BoxGeometry(), new MeshBasicMaterial()));
+    const cleanScene = sceneManager.getCleanScene();
+
+    expect(cleanScene).toBeInstanceOf(Scene);
+    expect(cleanScene.children.length).toBe(0);
+
+    expect(cleanScene.up.x).toBe(0);
+    expect(cleanScene.up.y).toBe(1);
+    expect(cleanScene.up.z).toBe(0);
+
+    expect(cleanScene.position.x).toBe(0);
+    expect(cleanScene.position.y).toBe(0);
+    expect(cleanScene.position.z).toBe(0);
+
+    expect(cleanScene.scale.x).toBe(1);
+    expect(cleanScene.scale.y).toBe(1);
+    expect(cleanScene.scale.z).toBe(1);
+  });
+
+  it('should remove a label from the scene', () => {
+    jest.spyOn(sceneManager, 'getObjectsGroup');
+    sceneManager.removeLabel('test');
+    expect(sceneManager.getObjectsGroup).toHaveBeenCalledTimes(1);
+  });
+
+  it('should add new types of objects (Jets, Tracks...) to the event data group', () => {
+    const eventGroup = sceneManager.addEventDataTypeGroup('object');
+
+    expect(eventGroup).toBeInstanceOf(Group);
+    expect(eventGroup.parent.name).toBe('EventData');
+    expect(eventGroup.parent.parent).toBeInstanceOf(Scene);
+    expect(eventGroup.parent.parent.children.length).toBe(6);
+    expect(eventGroup.userData).toEqual({});
+  });
+
+  it('should clear event data of the scene', () => {
+    jest.spyOn(sceneManager, 'getEventData');
+    sceneManager.clearEventData();
+    expect(sceneManager.getEventData).toHaveBeenCalledTimes(2);
```

Comment:
I think we should instead assert there that the scene does not contain the event data group. The assertion right now does not test the intended behavior.